taxi/forms.py

```diff
@@ -0,0 +1,44 @@
+from django import forms
+from django.contrib.auth import get_user_model
+from django.contrib.auth.forms import UserCreationForm
+from django.core.validators import RegexValidator
+
+from taxi.models import Car, Driver
+
+
+class LicenseNumberFormMixin(forms.Form):
+    license_number = forms.CharField(
+        max_length=8,
+        validators=[
+            RegexValidator(
+                regex=r"^[A-Z]{3}[0-9]{5}$",
+                message="The license number must contain 3 capital letters "
+                        "and 5 digits (for example, ABC12345)."
+            )
+        ]
+    )
+
+
+class DriverCreationForm(UserCreationForm, LicenseNumberFormMixin):
+    class Meta(UserCreationForm.Meta):
+        model = Driver
+        fields = UserCreationForm.Meta.fields + (
+            "first_name", "last_name", "license_number"
+        )
+
+
+class DriverLicenseUpdateForm(LicenseNumberFormMixin, forms.ModelForm):
+    class Meta:
+        model = Driver
+        fields = ("license_number",)
+
+
+class CarForm(forms.ModelForm):
+    drivers = forms.ModelMultipleChoiceField(
+        queryset=get_user_model().objects.all(),
+        widget=forms.CheckboxSelectMultiple,
+    )
+
+    class Meta:
+        model = Car
+        fields = "__all__"
```

Comment:
It's okey to use validation on `Forms` level, but this is best done at the model level. This way, we can make sure that invalid data does not get into our database in any way

[Documentation](https://docs.djangoproject.com/en/5.1/ref/forms/validation/#cleaning-a-specific-field-attribute)